File: android/MaskApp/app/src/main/java/kr/study/maskinfo/adapters/StoreAdapter.java
package kr.study.maskinfo.adapters;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import kr.study.maskinfo.R;
import kr.study.maskinfo.models.Store;

public class StoreAdapter extends RecyclerView.Adapter<StoreAdapter.StoreViewHolder> {
    private List<Store> mItems = new ArrayList<>();

    // 아이템 뷰의 정보를 가지고 있는 클래스
    static class StoreViewHolder extends RecyclerView.ViewHolder {
        TextView tvName;
        TextView tvAddr;
        TextView tvDistance;
        TextView tvRemain;
        TextView tvCount;

        public StoreViewHolder(@NonNull View itemView) {
            super(itemView);

            tvName = itemView.findViewById(R.id.tv_name);
            tvAddr = itemView.findViewById(R.id.tv_addr);
            tvDistance = itemView.findViewById(R.id.tv_distance);
            tvRemain = itemView.findViewById(R.id.tv_remain);
            tvCount = itemView.findViewById(R.id.tv_count);
        }
    }

    public void updateItems(List<Store> items) {
        mItems = items;
        notifyDataSetChanged(); // UI 갱신
    }

    @NonNull
    @Override
    public StoreViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_store, parent, false);
        return new StoreViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull StoreViewHolder holder, int position) {
        Store store = mItems.get(position);

        holder.tvName.setText(store.getName());
        holder.tvAddr.setText(store.getAddr());
        holder.tvDistance.setText("1.0km");

        String remainStat = "충분";
        String count = "100개 이상";
        int color = Color.GREEN;
        switch (store.getRemainStat()) {
            case "plenty": count = "100개 이상"; remainStat = "충분"; color = Color.GREEN; break;
            case "some": count = "30개 이상"; remainStat = "여유"; color = Color.YELLOW; break;
            case "few": count = "2개 이상"; remainStat = "매진 임박"; color = Color.RED; break;
            case "empty": count = "1개 이하"; remainStat = "재고 없음"; color = Color.GRAY; break;
            default: break;
        }

        holder.tvRemain.setText(remainStat);
        holder.tvCount.setText(count);
        holder.tvRemain.setTextColor(color);
        holder.tvCount.setTextColor(color);
    }

    @Override
    public int getItemCount() {
        return mItems.size();
    }
}

File: android/MaskInfoKt/app/src/main/java/kr/study/maskinfokt/StoreAdapter.kt
package kr.study.maskinfokt

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import kr.study.maskinfokt.models.Store
import java.util.*

//아이템 뷰 정보를 가지고 있는 클래스
class StoreViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    var tv_name: TextView = itemView.findViewById(R.id.textview_name)
    var tv_addr: TextView = itemView.findViewById(R.id.textview_addr)
    var tv_dist: TextView = itemView.findViewById(R.id.textview_dist)
    var tv_remain: TextView = itemView.findViewById(R.id.textview_remain)
    var tv_count: TextView = itemView.findViewById(R.id.textview_count)
}

class StoreAdapter : RecyclerView.Adapter<StoreViewHolder>() {
    private var mItems: List<Store> = ArrayList<Store>()
    fun updateItems(items: List<Store>) {
        mItems = items
        notifyDataSetChanged() //ui 갱
    }

    //뷰홀더를 만드는 부분
    //뷰홀더 클래스를 만들어주고 리턴시켜야함
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StoreViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(
            R.layout.item_store,
            parent,
            false
        ) //adapter가 항상 쓰는 패턴, 사 외워야하는건 아니고.. 개발자 홈페이지에 가이드 보고 하면 되는 것임.
        return StoreViewHolder(view)
    }

    //실제 데이터를 표시하는 부분
    override fun onBindViewHolder(holder: StoreViewHolder, position: Int) {
        val store: Store = mItems[position]
        holder.tv_name.text = store.name
        holder.tv_addr.text = store.addr
        holder.tv_dist.text = "1km"
        var count = "100개 이상"
        var remainStat = "충분"
        var color = Color.BLUE
        when (store.remain_stat) {
            "planty" -> {
                count = "100개 이상"
                remainStat = "충분"
                color = Color.GREEN
            }
            "some" -> {
                count = "30개 이상"
                remainStat = "여유"
                color = Color.YELLOW
            }
            "few" -> {
                count = "2개 이상"
                remainStat = "매진 임박"
                color = Color.RED
            }
            "empty" -> {
                count = "1개 이상"
                remainStat = "매진"
                color = Color.GRAY
            }
            else -> {
            }
        }
        holder.tv_remain.text = remainStat
        holder.tv_count.text = count
        holder.tv_remain.setTextColor(color)
        holder.tv_count.setTextColor(color)
    }

    //데이터의 갯수
    override fun getItemCount(): Int = mItems.size


}